SwiftGit2/References.swift

```diff
@@ -119,6 +120,27 @@ public struct Branch: ReferenceType {
 		}
 		commit = PointerTo<Commit>(oid)
 	}
+	
+	public func allCommits(in repo: Repository) -> [Commit] {
+		var commits: [Commit] = []
+		var walker: OpaquePointer? = nil
+		var unsafeCommit: OpaquePointer? = nil
+		var oid = self.oid.oid
+		git_revwalk_new(&walker, repo.pointer)
+		git_revwalk_sorting(walker, GIT_SORT_TOPOLOGICAL.rawValue)
+		git_revwalk_push(walker, &oid)
+		while git_revwalk_next(&oid, walker) == GIT_OK.rawValue {
+			guard
+				git_commit_lookup(&unsafeCommit, repo.pointer, &oid) == GIT_OK.rawValue,
+				let commit = unsafeCommit else {
+					return commits
+			}
+			commits += [Commit(commit)]
+			git_commit_free(unsafeCommit)
+		}
+		git_revwalk_free(walker)
```

Comment:
If the function returns on line 136, this `git_revwalk_free` will never be called and the object will leak. Perhaps move this to a `defer { ... }` statement just after the `git_revwalk_new`, to ensure it is called for every return path.